src/httpserver.cpp

```diff
@@ -239,6 +239,12 @@ static void http_request_cb(struct evhttp_request* req, void* arg)
         return;
     }
 
+    // Dump headers if request is valid
+    LogPrint(BCLog::HTTP, "Request headers:\n",
```

Comment:
it should avoid doing this if `!LogAcceptCategory(BCLog::HTTP)`, no need for the overhead to copy the headers when the result is ignored